config/routes.rb

```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  get '/users/:user_id/posts', to:'posts#index', as: :user_posts
+  get '/users/:user_id/posts/:id', to:'posts#show', as: :user_post
+  get '/users', to:'users#index', as: :users
+  get '/users/:id', to:'users#show', as: :user
```

Comment:
- [ ] [OPTIONAL] You did great adding the routing but I think it would be better to use resources
```ruby
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

Reply:
Thank you @hiromataba, you suggested me a good shortcut, I will use it.